### DISCUSSION TITLE
Exclude baking-related code from Unity runtime?

### DISCUSSION THREAD

**Discussion by laicasaane** [cathei/BakingSheet#24] [Q&A]
I want to only include the code that supports reading data in the final build. I don't want the build to contain any baking or reflection code. Could AOT code stripping help me with this if I never use baking code at runtime?

If AOT cannot help, would it be possible to restructure this library into a `BakingSheet.Runtime.asmdef` and a `BakingSheet.Baketime.asmdef`, so that we can exclude or include `BakingSheet.Baketime.asmdef` at will?

**Comment by cathei** [comment]
Good question - Higher AOT code stripping level should exclude the code that is not being used by your main assembly. Additionally, non-Json converters are included in Editor only by default.

Can you give me more context though, what would be the primary reason of excluding reflection code? BakingSheet (and Json.NET) do need reflection to properly set values to your properties.

**Reply by laicasaane** [reply]
I only use BakingSheet at editing time to export Excels to ScriptableObjects. At runtime, I only retrieve readonly data from the SheetContainer. There won't be runtime conversion of any kind.

**Reply by cathei** [reply]
Alright. If you are converting it to own ScriptableObject, then there would be no reference to BakingSheet, so in that case AOT stripper should strip unused code out.
